tests/system/common.py

```diff
@@ -328,14 +328,13 @@ def assert_app_tasks_healthy(client, app_def):
 def get_marathon_leader_not_on_master_leader_node():
     marathon_leader = shakedown.marathon_leader_ip()
     master_leader = shakedown.master_leader_ip()
-    print('marathon: {}'.format(marathon_leader))
-    print('leader: {}'.format(master_leader))
+    print('marathon leader: {}'.format(marathon_leader))
+    print('mesos leader: {}'.format(master_leader))
 
     if marathon_leader == master_leader:
         delete_marathon_path('v2/leader')
         shakedown.wait_for_service_endpoint('marathon', timedelta(minutes=5).total_seconds())
-        marathon_leadership_changed(marathon_leader)
-        marathon_leader = shakedown.marathon_leader_ip()
```

Comment:
this is where the extra call to mesos DNS was made and is no longer there...

Reply:
This is a little confusing code. Isn't `marathon_leadership_changed` calling the DNS as well?

Reply:
well the thing is there are multiple masters and multiple mesosDNS instances. By querying multiple times we might hit different masters. On every master, mesosDNS is updated every 30s but this does not have to be synchronized in between masters...

another approach might be to explicitly wait for the information being propagated to all masters... 

but we are just trying to verify that the election is finished so maybe the fact that it got propagated to one master is enough from the test perspective... (but still different masters might return different info at that point...)